src/main/java/duke/tasks/Deadline.java

```diff
@@ -0,0 +1,64 @@
+package duke.tasks;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Represents the task with deadline specifications.
+ */
+public class Deadline extends Task {
+
+    protected Date by;
+
+    /**
+     * Create an initial deadline task that initializes
+     * to incomplete by default.
+     * @param description a description given for the task
+     * @param by the actual deadline for the deadline task
+     */
+    public Deadline(String description, Date by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Creates a deadline task mainly for storage purposes.
+     * This constructor keeps the state of the task whether
+     * is it done or incomplete.
+     *
+     * @param description a description given for the task
+     * @param by the date and time which the task needs to be
+     *           completed by
+     * @param doner the completion status of the task
+     *              (Completed is 1, incomplete is 2)
+     */
+    public Deadline(String description, Date by, int doner) {
```

Comment:
doner can be renamed to isDone
and if the status can only be 1/2, change to Boolean type.